frontend/src/components/contactsForm.tsx

```diff
@@ -0,0 +1,310 @@
+      return
+    }
+    setRelationships((prevRelationships) => [
+      ...prevRelationships,
+      {
+        relationshipType
+      }
+    ])
+    setRelationshipType('')
+  }
+
+  const handleRelationshipChange = (
+    index: number,
+    field: keyof Relationship,
+    value: any
+  ) => {
+    setRelationships((prevRelationships) =>
+      prevRelationships.map((type, i) =>
+        i === index ? { ...type, [field]: value } : type
+      )
+    )
+  }
+
+  const handleRemoveRelationshipType = (index: number) => {
+    setRelationships((prevRelationships) => [
+      ...prevRelationships.slice(0, index),
+      ...prevRelationships.slice(index + 1)
+    ])
+  }
+
+  const addImportantEvent = () => {
+    if (eventDate.trim() === '' || eventType.trim() === '') {
+      setShowError(true)
+      return
+    }
+    setImportantDates((prevDates) => [
+      ...prevDates,
+      {
+        date: eventDate,
+        eventType
+      }
+    ])
+    setEventDate('')
+    setEventType('')
+  }
+
+  const handleImportantEventChange = (
+    index: number,
+    field: keyof ImportantDate,
+    value: any
+  ) => {
+    setImportantDates((prevDates) =>
+      prevDates.map((date, i) =>
+        i === index ? { ...date, [field]: value } : date
+      )
+    )
+  }
+
+  const handleRemoveImportantEvent = (index: number) => {
+    setImportantDates((prevDates) => [
+      ...prevDates.slice(0, index),
+      ...prevDates.slice(index + 1)
+    ])
+  }
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      // TODO(FK) T144: change url once app deployed
+      await axios.post('http://localhost:8000/contacts', {
+        firstName,
+        lastName,
+        relationships,
+        importantDates,
+        email,
+        phoneNumber,
+        notes
+      }, {
+        withCredentials: true
+      })
+      handleContactCreate({
+        firstName,
+        lastName,
+        relationships,
+        importantDateEvent: importantDates,
+        email,
+        phoneNumber,
+        notes
+      })
+    } catch (error) {
+      console.error(error)
+      setErrorMessage((error as any).response.data ?? 'Unknown error occured.')
+    }
+  }
+
+  return (
+    <div className="contacts_form">
+      {showForm && (
+        <>
+          <form onSubmit={handleSubmit}>
+            <button type='button' onClick={handleClose}>X</button>
+            {(errorMessage.length > 0) && <p className="error_message">{errorMessage}</p>}
+            {<p className="show_error">{showError}</p>}
+            <div className="user_input">
+              <label htmlFor="first_name">First name</label>
+              <input
+                className="first_name"
+                name="first_name"
+                type="text"
+                autoComplete="on"
+                required
+                value={firstName}
+                onChange={handleFirstNameChange}
+              ></input>
```

Comment:
if you don't have any child elements to nest, I think this is just `<input ... />` and you don't need a pair of tags 


```
              <input
                className="first_name"
                name="first_name"
                type="text"
                autoComplete="on"
                required
                value={firstName}
                onChange={handleFirstNameChange}
              />
```